forest/src/cli/wallet_cmd.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::stringify_rpc_err;
+use rpc_client::{balance, default, export, list, new, new_client, set_default, sign};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum WalletCommands {
+    /// Prints out wallet balance of specified address via RPC
+    #[structopt(about = "Get account balance")]
+    Balance {
+        #[structopt(short, help = "Input a valid address")]
+        address: String,
+    },
+
+    /// Creates wallet of specified signature type via RPC
+    #[structopt(
+        about = "Generate a new key of the given type",
+        help = "Generate a new key of the given type"
+    )]
+    New {
+        #[structopt(
+            short,
+            default_value = "secp256k1",
+            help = "Input signature type [bls | secp256k1] (default secp256k1)"
+        )]
+        sig_type: String,
+    },
+
+    /// Prints out all addresses in the wallet via RPC
+    #[structopt(about = "List wallet address", help = "List wallet address")]
+    List,
+
+    /// Marks the given address as as the default wallet via RPC
+    #[structopt(
+        about = "Set default wallet address",
+        help = "Set default wallet address"
+    )]
+    SetDefault {
+        #[structopt(short, help = "Input valid address")]
+        address: String,
+    },
+
+    /// Prints out the address marked as default in the wallet via RPC
+    #[structopt(
+        name = "default",
+        about = "Get default wallet address",
+        help = "Get default wallet address"
+    )]
+    Def,
+
+    /// Prints out private key of specified address in wallet via RPC
+    #[structopt(about = "Exports wallet keys", help = "Exports wallet keys")]
+    Export {
+        #[structopt(short, help = "Input valid address")]
+        address: String,
+    },
+
+    /// Signs the given bytes using the given address via RPC
+    #[structopt(about = "Sign a message", help = "Sign a message")]
+    Sign {
+        #[structopt(short, help = "Must specify signing address")]
+        address: String,
+        #[structopt(short, help = "Must specify message to sign")]
+        message: String,
+    },
+}
+
+impl WalletCommands {
+    pub async fn run(&self) {
+        // TODO add verify and import cmds
+        match self {
+            Self::Balance { address } => {
+                let mut client = new_client();
+
+                let balance = balance(&mut client, address.to_string())
+                    .await
+                    .map_err(stringify_rpc_err)
```

Comment:
dont see the point of mapping the error if your arent going to use it

Reply:
i think the map_err should stay, the json_rpc_error does not implement debug so the error needs to be stringified before unwrap